src/main/java/org/eclipse/xpanse/terraform/boot/terraform/service/TerraformTaskResultHelper.java

```diff
@@ -0,0 +1,120 @@
+import org.eclipse.xpanse.terraform.boot.models.exceptions.FailedCallbackResponseStoreException;
+import org.eclipse.xpanse.terraform.boot.models.exceptions.ResultAlreadyReturnedOrRequestIdInvalidException;
+import org.eclipse.xpanse.terraform.boot.models.response.TerraformResult;
+import org.eclipse.xpanse.terraform.boot.utils.SerializerUtils;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+/**
+ * Terraform service classes are manage task result.
+ */
+@Slf4j
+@Component
+public class TerraformTaskResultHelper {
+
+    private static final String TF_RESULT_FILE_SUFFIX = ".dat";
+
+    @Value("${failed.callback.response.store.location}")
+    private String failedCallbackStoreLocation;
+
+    @Resource
+    private SerializerUtils serializerUtils;
+
+    /**
+     * When the terraform-boot callback fails, store the TerraformResult in the local file system.
+     *
+     * @param result TerraformResult.
+     */
+    public void storeFailedCallbackResponse(TerraformResult result) {
+        String filePath = getFilePath(result.getRequestId());
+        File file = new File(filePath);
+        if (!file.exists() && !file.mkdirs()) {
+            throw new FailedCallbackResponseStoreException("Failed to create directory: "
+                    + filePath);
+        }
+        Serializer<byte[]> serializer = serializerUtils
+                .getSerializerByFoundation(TerraformResult.class);
+        byte[] terraformResultData = serializer.serialize(result);
+        try (FileOutputStream fos = new FileOutputStream(file.getPath() + File.separator
+                        + result.getRequestId() + TF_RESULT_FILE_SUFFIX)) {
+            fos.write(terraformResultData);
+            log.info("Store failed callback response success, fileName: {}",
+                    result.getRequestId());
+        } catch (IOException e) {
+            log.error("Store failed callback response failure, fileName: {}",
+                    result.getRequestId(), e);
+            throw new FailedCallbackResponseStoreException("Store failed callback response failure,"
+                    + e);
+        }
+    }
+
+    /**
+     * Get the TerraformResult object stored when the terraform-boot callback fails by RequestId.
+     *
+     * @param requestId requestId.
+     * @return TerraformResult.
+     */
+    public TerraformResult getFailedCallbackResponseByRequestId(String requestId) {
+        String filePath = getFilePath(UUID.fromString(requestId));
+        File resultFile = new File(filePath + File.separator + requestId + TF_RESULT_FILE_SUFFIX);
+        if (!resultFile.exists() && !resultFile.isFile()) {
+            throw new ResultAlreadyReturnedOrRequestIdInvalidException(
+                    "Result file does not exist: " + resultFile.getAbsolutePath());
+        }
+        try (FileInputStream fis = new FileInputStream(resultFile)) {
+            Serializer<byte[]> serializer = serializerUtils
```

Comment:
we must move this entire logic to the SerializerUtils. 

Just send the byte[] to 'deserialize' method in SerializerUtil and get the TerraformResult.